src/vs/editor/browser/viewParts/lineNumbers/lineNumbers.ts

```diff
@@ -57,7 +61,15 @@ export class LineNumbersOverlay extends DynamicViewOverlay {
 		return true;
 	}
 	public onCursorPositionChanged(e:editorCommon.IViewCursorPositionChangedEvent): boolean {
-		return false;
+		let modelPosition = this._context.model.convertViewPositionToModelPosition(e.position.lineNumber, e.position.column);
+
+		if (!this._relativeLineNumbers || this._currentLineNumber === modelPosition.lineNumber) {
+			return false;
+		}
+
+		this._currentLineNumber = modelPosition.lineNumber;
```

Comment:
In the case where `this._relativeLineNumbers` is `false`, `this._currentLineNumber` should still be assigned such that `this._currentLineNumber` always has a good value.

It is possible that the `relativeLineNumbers` option changes at runtime and in that case we'd want `this._currentLineNumber` to be up-to-date.